### PR TITLE
Adding business owner APIs

### PR DESCRIPTION
**Pull request by lakshani**

### CODE REVIEW

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 1]
```diff
@@ -152,7 +152,7 @@ public BusinessOwner getBusinessOwner(int businessOwnerId) throws AppManagementE
 
         Connection connection = null;
         PreparedStatement statementToGetBusinessOwners = null;
-        BusinessOwner businessOwner = new BusinessOwner();
+        BusinessOwner businessOwner = null;
         ResultSet businessOwnerResultSet = null;
 
         String queryToGetBusinessOwner = "SELECT OWNER_NAME, OWNER_EMAIL, OWNER_DESC, OWNER_SITE FROM BUSINESS_OWNER " +
```
Comment (lahiru823) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:158]: BUSINESS_OWNER add the prefix APM

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 2]
```diff
@@ -23,44 +23,230 @@
...
     private static final Log log = LogFactory.getLog(AdministrationApiServiceImpl.class);
     BeanValidator beanValidator;
 
     @Override
     public Response administrationBusinessownerGet(String accept, String ifNoneMatch) {
-        return null;
+        BusinessOwnerListDTO businessOwnerListDTO = new BusinessOwnerListDTO();
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            //get business owner related to id.
```
Comment (rushmin) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:53]: Comment says about an ID. But there is no ID in the code.

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 3]
```diff
@@ -23,44 +23,230 @@
...
+            for (BusinessOwner businessOwner : businessOwners) {
+                BusinessOwnerDTO businessOwnerDTO = new BusinessOwnerDTO();
+                businessOwnerDTO.setId(businessOwner.getBusinessOwnerId());
+                businessOwnerDTO.setName(businessOwner.getBusinessOwnerName());
+                businessOwnerDTO.setEmail(businessOwner.getBusinessOwnerEmail());
+                businessOwnerDTO.setDescription(businessOwner.getBusinessOwnerDescription());
+                businessOwnerDTO.setSite(businessOwner.getBusinessOwnerSite());
+                businessOwnerPropertyList = businessOwner.getBusinessOwnerPropertiesList();
+                List<BusinessOwnerPropertiesDTO> businessOwnerPropertiesDTOList = new ArrayList<>();
+                //save custom properties of the owner.
```
Comment (rushmin) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:70]: Invalid comment.

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 4]
```diff
@@ -23,44 +23,230 @@
...
+        beanValidator = new BeanValidator();
+        beanValidator.validate(body);
+        BusinessOwnerDTO businessOwnerDTO = new BusinessOwnerDTO();
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            String ownerName = body.getName();
+            String ownerEmail = body.getEmail();
+            String ownerDescription = body.getDescription();
+            String ownerSite = body.getSite();
+            if (StringUtils.isEmpty(ownerName)) {
```
Comment (rushmin) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:100]: Use "Apache commons" StringUtils

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 5]
```diff
@@ -23,44 +23,230 @@
...
+            businessOwnerListDTO.setBusinessOwnerList(businessOwnerDTOList);
+        } catch (AppManagementException e) {
+            String errorMessage = "Error while retrieving business owners.";
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return Response.ok().entity(businessOwnerListDTO).build();
     }
 
     @Override
     public Response administrationBusinessownerPost(BusinessOwnerDTO body, String contentType, String ifModifiedSince) {
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:90]: body >> businessOwnerDTO

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 6]
```diff
@@ -23,44 +23,230 @@
...
+        }
+        return Response.ok().build();
     }
 
     @Override
     public Response administrationBusinessownerBusinessOwnerIdDelete(String businessOwnerId, String ifMatch, String ifUnmodifiedSince) {
-        return null;
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            //delete the business owner.
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:243]: Change comments. Start with Capital, end with full stop like a sentence

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 7]
```diff
@@ -23,44 +23,230 @@
...
+            apiProvider.deleteBusinessOwner(businessOwnerId);
+        } catch (AppManagementException e) {
+            ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getMessage(), 500l, e.getCause().getMessage());
```
Comment (rushmin) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:246]: No need to construct an ErrorDTO. Rather we can use RestApiUtil.handleInternalServerError

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 1 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:158]: Fixed in latest code. 
https://github.com/wso2/carbon-appmgt/commit/c944485fa42a4901011334d0918ee7561a6554a2#diff-208788808ce86c680d57a6cdba990a9fR165

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 8]
```diff
@@ -23,44 +23,230 @@
...
+                    RestApiUtil.handleBadRequest("Business owner properties cannot be empty or null.", log);
+                }
+                businessOwnerProperty.setPropertyId(propertyId.trim());
+                businessOwnerProperty.setPropertyValue(propertyValue.trim());
+                businessOwnerProperty.setShowingInStore(isVisible);
+                businessOwnerPropertyList.add(businessOwnerProperty);
+            }
+            businessOwner.setBusinessOwnerPropertiesList(businessOwnerPropertyList);
+
+            //save business owner.
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:128]: Start comment with a capital letter

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 9]
```diff
@@ -23,44 +23,230 @@
...
+            BusinessOwner addedBusinessOwner = apiProvider.getBusinessOwner(ownerId);
+            businessOwnerDTO.setId(ownerId);
+            businessOwnerDTO.setName(addedBusinessOwner.getBusinessOwnerName());
+            businessOwnerDTO.setEmail(addedBusinessOwner.getBusinessOwnerEmail());
+            businessOwnerDTO.setDescription(addedBusinessOwner.getBusinessOwnerDescription());
+            businessOwnerDTO.setSite(addedBusinessOwner.getBusinessOwnerSite());
+
+            List<BusinessOwnerProperty> ownerPropertyList = businessOwner.getBusinessOwnerPropertiesList();
+            List<BusinessOwnerPropertiesDTO> ownerPropertiesDTOList = new ArrayList<>();
+            //save custom properties of the owner.
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:141]: Start comment with a capital letter

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 10]
```diff
@@ -23,44 +23,230 @@
...
+        return Response.ok().entity(businessOwnerDTO).build();
     }
 
     @Override
     public Response administrationBusinessownerBusinessOwnerIdGet(String businessOwnerId, String accept, String ifNoneMatch) {
-        return null;
+        BusinessOwnerDTO businessOwnerDTO = new BusinessOwnerDTO();
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            //get policy details related to id
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:162]: Invalid comment.

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java [hunk 11]
```diff
@@ -23,44 +23,230 @@
...
+            String errorMessage = "Error while retrieving details of business owner Business owner Id : " +
+                    businessOwnerId;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return Response.ok().entity(businessOwnerDTO).build();
     }
 
     @Override
     public Response administrationBusinessownerBusinessOwnerIdPut(String businessOwnerId, BusinessOwnerDTO body, String contentType, String ifMatch, String ifUnmodifiedSince) {
-        return null;
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:58]: body >> businessOwnerId

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 2 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:53]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 3 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:70]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 5 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:90]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 4 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:100]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 8 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:128]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 9 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:141]: This comment is removed by someone in latest code.

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 10 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:162]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 11 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:58]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 6 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:243]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

File: components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java (hunk 7 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java:246]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0

**Review by lakshani — COMMENTED** [review]
Added changes based on code review comments.

**Pull request merged**